tests/modules/get_professors_by_class/app/get_professors_by_class_usecase.test.ts

```diff
@@ -11,32 +12,41 @@ describe('GetProfessorsByClassUsecase', () => {
     usecase = new GetProfessorsByClassUsecase(scheduleRepository);
   });
 
-  it('should return a list of professors for a valid classId', async () => {
-    const classId = '0a8c5357-1f07-5b24-9845-9318c47ab923'; 
+  it('should return a list of professors with availabilities for a valid classId', async () => {
+    const classId = '0a8c5357-1f07-5b24-9845-9318c47ab923';
     const result = await usecase.execute(classId);
 
     expect(result).toBeInstanceOf(Array);
     expect(result.length).toBeGreaterThan(0);
+
     result.forEach(professor => {
-      expect(professor.role).toBe('PROFESSOR');
       expect(professor).toHaveProperty('name');
       expect(professor).toHaveProperty('email');
       expect(professor).toHaveProperty('RA');
+
+      expect(professor).toHaveProperty('availabilities');
+      expect(professor.availabilities).toBeInstanceOf(Array);
+
+      professor.availabilities.forEach(availability => {
+        expect(availability).toHaveProperty('availabilityId');
+        expect(availability).toHaveProperty('weekDay');
+        expect(availability).toHaveProperty('startTime');
+        expect(availability).toHaveProperty('endTime');
+        expect(availability).toHaveProperty('isTaken');
+      });
     });
   });
 
-  it('should be empty array if no professors are suitable for the class', async () => {
-    const classId = '0a8c5357-1f07-5b24-9845-9318c47ab9aa'; 
+  it('should return an empty array if no professors are suitable for the class', async () => {
+    const classId = '0a8c5357-1f07-5b24-9845-9318c47ab9aa';
     const result = await usecase.execute(classId);
     expect(result).toBeInstanceOf(Array);
     expect(result.length).toBe(0);
-
   });
 
-  it('should throw an error if classId is invalid', async () => {
+  it('should throw a NoItemsFound error if classId is invalid', async () => {
```

Comment:
If the classId iis not valid it should return a EntityErro(BadRequest) not a NoItemsFound, NoItemsFound should only be return if the ID is valid but it does not have any object for this value on the repo